src/components/common/ColonyHome/ColonyUnclaimedTransfers/ColonyUnclaimedTransfers.tsx

```diff
@@ -42,31 +41,58 @@ const MSG = defineMessages({
 });
 
 const ColonyUnclaimedTransfers = () => {
-  const { colony } = useColonyContext();
-  const { colonyAddress, name } = colony || {};
-
-  const { data, error } = useColonyTransfersQuery({
-    variables: { address: colonyAddress },
+  const { colony, canInteractWithColony } = useColonyContext();
+  const { colonyAddress, name, fundsClaims, chainFundsClaim, tokens } =
+    colony || {};
+  const { items: claims = [] } = fundsClaims || {};
+
+  /*
+   * @NOTE We have to do some very heavy lifting (more or less) here due to us
+   * not being able to use Apollo's cache, so we want to short-circuit early
+   * in order to not waste any computing resources unecesarily
+   */
+  if (!chainFundsClaim && !fundsClaims) {
+    return null;
+  }
+
+  const chainClaimWithToken = chainFundsClaim
+    ? {
+        ...chainFundsClaim,
+        token: tokens?.items?.find(
+          (token) => token?.token?.tokenAddress === ADDRESS_ZERO,
+        )?.token,
+      }
+    : null;
+
+  /*
+   * Claims data needs to be merged, both ERC20's and Native Chain Tokens
+   *
+   * Also, we have to sort in-client since Apollo's cache is being a bitch
+   * not allowing more than 3 local field entries without breaking, forcing us
+   * to do the sorting / merging here, rather than at the time we fetch data.
+   * This kinda sucks!
+   */
+  const sortedFundsClaims = [...claims, chainClaimWithToken]
+    .filter((claim) => !!claim)
```

Comment:
We have an util method for that: `notNull` in `~utils/arrays`

Reply:
I know, and I preferred to not use it :P I think I even removed it in some place. But can't really remember my reasoning back then for it.

Anyway, I'll refactor this to make use of it.